operator/v1/types_dns.go

```diff
@@ -163,6 +173,25 @@ var (
 	DNSLogLevelTrace DNSLogLevel = "Trace"
 )
 
+// DNSNodeService represents a Kubernetes service by name and namespace for node services.
+type DNSNodeService struct {
+	// name is the name of the service.
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=`^[a-z]([-a-z0-9]*[a-z0-9])?$`
+	Name string `json:"name"`
+
+	// namespace is the namespace of the service.
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=`^[a-z]([-a-z0-9]*[a-z0-9])?$`
```

Comment:
Prefer CEL to pattern here
```suggestion
	// +kubebuilder:validation:XValidation:rule=`!format.dns1123Label().validate(self).hasValue()`,message="the value must consist of only lowercase alphanumeric characters and hyphens"
```